### ISSUE TITLE
-[FFYDaemonController checkIfDaemonIsRunning:]: unrecognized selector sent to instance

### ISSUE THREAD

**Issue by dainkaplan** [ivanvc/mongodb-prefpane#2]
I'm getting the following error when I try to toggle the daemon on. I might look into it myself, but i'm posting an issue for it anyways, since it doesn't seem to be working. I'm using the version from downloads 0.4.

```
3/12/12 12:22:18.733 PM System Preferences: -[FFYDaemonController checkIfDaemonIsRunning:]: unrecognized selector sent to instance 0x400468900
3/12/12 12:22:18.733 PM System Preferences: -[FFYDaemonController checkIfDaemonIsRunning:]: unrecognized selector sent to instance 0x400468900
3/12/12 12:22:18.734 PM System Preferences: (
    0   CoreFoundation                      0x00007fff84d16fc6 __exceptionPreprocess + 198
    1   libobjc.A.dylib                     0x00007fff86e98d5e objc_exception_throw + 43
    2   CoreFoundation                      0x00007fff84da32ae -[NSObject doesNotRecognizeSelector:] + 190
    3   CoreFoundation                      0x00007fff84d03e73 ___forwarding___ + 371
    4   CoreFoundation                      0x00007fff84d03c88 _CF_forwarding_prep_0 + 232
    5   Foundation                          0x00007fff8d9c5f64 __NSFireTimer + 102
    6   CoreFoundation                      0x00007fff84ccbc24 __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__ + 20
    7   CoreFoundation                      0x00007fff84ccb776 __CFRunLoopDoTimer + 534
    8   CoreFoundation                      0x00007fff84cac001 __CFRunLoopRun + 1617
    9   CoreFoundation                      0x00007fff84cab676 CFRunLoopRunSpecific + 230
    10  HIToolbox                           0x00007fff9185731f RunCurrentEventLoopInMode + 277
    11  HIToolbox                           0x00007fff9185e5c9 ReceiveNextEventCommon + 355
    12  HIToolbox                           0x00007fff9185e456 BlockUntilNextEventMatchingListInMode + 62
    13  AppKit                              0x00007fff8bad0f5d _DPSNextEvent + 659
    14  AppKit                              0x00007fff8bad0861 -[NSApplication nextEventMatchingMask:untilDate:inMode:dequeue:] + 135
    15  AppKit                              0x00007fff8bacd19d -[NSApplication run] + 470
    16  AppKit                              0x00007fff8bd4bb88 NSApplicationMain + 867
    17  System Preferences                  0x0000000101e71d9c System Preferences + 7580
)
```

**Comment by dainkaplan** [comment]
I checked out the project, and after acquiring the DaemonController dependency separately (files were missing in Xcode project) and building, the newly compiled prefPane seems to work fine. Maybe just a problem with the 0.4 download?

**Comment by ivanvc** [comment]
Interesting... I'm trying to replicate the issue (using the 0.4 download), but not being able to do so. Which MacOS X version are you using? is it Lion? I'm running 10.7.2.

Is the issue happening right away when you open the prefpane? or when you click on the ON/OFF button?

Thanks for taking a look and giving a try to this project :)

**Comment by dainkaplan** [comment]
I noticed that those errors are still appearing in the log, so it wasn't related to it not working originally. It looks like it didn't like me specifying ~/ in the path to my config file originally (which I also changed to an absolute path at some point in trying to get it to work). Sorry for the confusion. You might want to think about adding ~/ expansion to the arguments. 

(I am also running Lion)
